branchpro/new_models.py

```diff
@@ -0,0 +1,332 @@
+
+        # Invert order of serial intervals for ease in _normalised_daily_mean
+        self._serial_interval = np.asarray(serial_interval)[::-1]
+        self._r_profile = np.array([initial_r])
+        self._overdispesion = phi
+        self._normalizing_const = np.sum(self._serial_interval)
+
+    def set_overdispersion(self, phi):
+        """
+        Updates overdispersion noise parameter for the model.
+
+        Parameters
+        ----------
+        phi
+            New value of the overdispersion parameter for the negative
+            binomial noise distribution.
+
+        """
+        if not isinstance(phi, (int, float)):
+            raise TypeError(
+                'Value of overdispersion must be integer or float.')
+        if phi <= 0:
+            raise ValueError(
+                'Value of overdispersion must be must be > 0. For \
+                overdispesion = 0, please use `BranchProModel` class type.')
+
+        self._overdispesion = phi
+
+    def get_overdispersion(self):
+        """
+        Returns overdispersion noise parameter for the model.
+
+        """
+        return self._overdispesion
+
+    def simulate(self, parameters, times):
+        """
+        Runs a forward simulation with the given ``parameters`` and returns a
+        time-series with incidence numbers corresponding to the given ``times``
+        .
+
+        Parameters
+        ----------
+        parameters
+            Initial number of cases.
+        times
+            The times at which to evaluate. Must be an ordered sequence,
+            without duplicates, and without negative values.
+            All simulations are started at time 0, regardless of whether this
+            value appears in ``times``.
+
+        """
+        initial_cond = parameters
+        last_time_point = np.max(times)
+
+        # Repeat final r if necessary
+        # (r_1, r_2, ..., r_t)
+        if len(self._r_profile) < last_time_point:
+            missing_days = last_time_point - len(self._r_profile)
+            last_r = self._r_profile[-1]
+            repeated_r = np.full(shape=missing_days, fill_value=last_r)
+            self._r_profile = np.append(self._r_profile, repeated_r)
+
+        incidences = np.empty(shape=last_time_point + 1)
+        incidences[0] = initial_cond
+
+        # Construct simulation times in steps of 1 unit time each
+        simulation_times = np.arange(start=1, stop=last_time_point+1, step=1)
+
+        # Compute normalised daily means for full timespan
+        # and draw samples for the incidences
+        for t in simulation_times:
+            norm_daily_mean = self._r_profile[t-1] * (
+                self._effective_no_infectives(t, incidences))
+            if norm_daily_mean != 0:
+                nu = self._overdispesion * norm_daily_mean
+                incidences[t] = nbinom.rvs(
+                    n=nu * norm_daily_mean,
+                    p=1 / (1 + nu),
+                    size=1)
```

Comment:
Change it to be the same as the local imported version below, which is the one I think we got working